tests/test_json_to_parquet.py

```diff
@@ -0,0 +1,531 @@
+                        }
+                    ]
+                }),
+        )
+        iam_client.attach_role_policy(
+                RoleName=role_name,
+                PolicyArn=glue_service_policy_arn
+        )
+        iam_client.attach_role_policy(
+                RoleName=role_name,
+                PolicyArn=s3_read_policy_arn
+        )
+        yield glue_crawler_role["Role"]["Arn"]
+        iam_client.detach_role_policy(
+                RoleName=role_name,
+                PolicyArn=glue_service_policy_arn
+        )
+        iam_client.detach_role_policy(
+                RoleName=role_name,
+                PolicyArn=s3_read_policy_arn
+        )
+        iam_client.delete_role(RoleName=role_name)
+
+    @pytest.fixture()
+    def glue_crawler(self, glue_database, glue_database_name, glue_database_path, glue_flat_table,
+                     glue_flat_table_name, glue_nested_table, glue_nested_table_name,
+                     glue_crawler_role, json_s3_objects, namespace):
+        glue_client = boto3.client("glue")
+        crawler_name = f"{namespace}-pytest-crawler"
+        time.sleep(10) # give time for the IAM role trust policy to set in
+        glue_crawler = glue_client.create_crawler(
+                Name=crawler_name,
+                Role=glue_crawler_role,
+                DatabaseName=glue_database_name,
+                Description="A crawler for pytest unit test data.",
+                Targets={
+                    "S3Targets": [
+                        {
+                            "Path": os.path.join(
+                                glue_database_path,
+                                glue_flat_table_name.replace("_", "=", 1)
+                            ) + "/"
+                        },
+                        {
+                            "Path": os.path.join(
+                                glue_database_path,
+                                glue_nested_table_name.replace("_", "=", 1)
+                            ) + "/"
+                        }
+                    ]
+                },
+                SchemaChangePolicy={
+                    "DeleteBehavior": "LOG",
+                    "UpdateBehavior": "LOG"
+                },
+                RecrawlPolicy={
+                    "RecrawlBehavior": "CRAWL_NEW_FOLDERS_ONLY"
+                },
+                Configuration=json.dumps({
+                    "Version":1.0,
+                    "CrawlerOutput": {
+                        "Partitions": {
+                            "AddOrUpdateBehavior":"InheritFromTable"
+                        }
+                    },
+                    "Grouping": {
+                        "TableGroupingPolicy":"CombineCompatibleSchemas"
+                    }
+                })
+        )
+        glue_client.start_crawler(Name=crawler_name)
+        response = {"Crawler": {}}
+        for i in range(60): # wait up to 10 minutes for crawler to finish
```

Comment:
nit: should this be a `while` loop instead?

Reply:
If there is an issue on AWS end with the crawler I don't want the tests to hang forever. This is a timeout. Using a while loop is also possible but would require additional logic to both check for the condition and keep a counter for the timeout (`while crawler_not_done and my_counter < timout`)

Reply:
Got it. This makes sense!